Build failure: protonmail-bridge-gui-3.21.2 fails to build when trying to use Qt 6.10.0 as dependency

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

Package protonmail-bridge-gui-3.21.2 fails to build when trying to use Qt 6.10.0 as dependency.
If it helps, I'm on GNOME (not native QT environment).

### Can Hydra reproduce this build failure?

Yes

### Link to Hydra build job

https://hydra.nixos.org/build/310652797

### Relevant log output

```console
-- Found WrapOpenGL: TRUE
-- Found WrapVulkanHeaders: /nix/store/icyb0ffm7i4sajzphnf818firkqlx71h-vulkan-headers-1.4.328.0/include
-- Using Qt 6.10.0
CMake Error at /nix/store/r7wdw4nkkjii690wwnzkgj68cg4shh2v-qtbase-6.10.0/lib/cmake/Qt6Core/Qt6CoreMacros.cmake:3797 (message):
  Unexpected arguments: FILENAME_VARIABLE;deploy_script
Call Stack (most recent call first):
  /nix/store/r7wdw4nkkjii690wwnzkgj68cg4shh2v-qtbase-6.10.0/lib/cmake/Qt6Core/Qt6CoreMacros.cmake:3910 (qt6_generate_deploy_app_script)
  bridge-gui/CMakeLists.txt:175 (qt_generate_deploy_app_script)


-- Configuring incomplete, errors occurred!
error: Cannot build '/nix/store/kng2g8xhmpgwksyj6rpmjwpvjkw9fl5m-protonmail-bridge-gui-3.21.2.drv'.
       Reason: builder failed with exit code 1.
       Output paths:
         /nix/store/qcw3rz4h9g47w6gpmvw7j92qgic2dhsg-protonmail-bridge-gui-3.21.2
       Last 25 log lines:
       > -- Found OpenSSL: /nix/store/llswcygvgv9x2sa3z6j7i0g5iqqmn5gn-openssl-3.6.0/lib/libcrypto.so (found version "3.6.0")
       > -- Found c-ares: /nix/store/sa2l6zliibj07zchg9f7ggcz5jk01dr3-c-ares-1.34.5-dev/lib/cmake/c-ares/c-ares-config.cmake (found version "1.34.5")
       > -- Found RE2 via CMake.
       > -- Using gRPC 1.75.1
       > -- protoc found /nix/store/izjrnd79npwzbdcjiv0i36dcw3kigny9-protobuf-32.1/bin/protoc
       > -- grpc_cpp_plugin
       > -- grpc_cpp_plugin found at /nix/store/dvq2p53l0s7g3flya31d1ln0jdsahp2s-grpc-1.75.1/bin/grpc_cpp_plugin
       > -- Found GTest: /nix/store/dirw81fxgq6r441gpivg8z4x8k3rbnxs-gtest-1.17.0-dev/lib/cmake/GTest/GTestConfig.cmake (found version "1.17.0")
       > -- App name is Proton
       > -- App vendor is Proton
       > -- Bridge version is 3.21.2
       > -- Found qmake at /nix/store/r7wdw4nkkjii690wwnzkgj68cg4shh2v-qtbase-6.10.0/bin/qmake
       > -- Using Default install path (/nix/store/qcw3rz4h9g47w6gpmvw7j92qgic2dhsg-protonmail-bridge-gui-3.21.2), export BRIDGE_INSTALL_PATH to change it.
       > -- Found OpenGL: /nix/store/as2rg79vk4z7frb7hy27nmadc3fp1a4j-libglvnd-1.7.0/lib/libOpenGL.so
       > -- Found WrapOpenGL: TRUE
       > -- Found WrapVulkanHeaders: /nix/store/icyb0ffm7i4sajzphnf818firkqlx71h-vulkan-headers-1.4.328.0/include
       > -- Using Qt 6.10.0
       > CMake Error at /nix/store/r7wdw4nkkjii690wwnzkgj68cg4shh2v-qtbase-6.10.0/lib/cmake/Qt6Core/Qt6CoreMacros.cmake:3797 (message):
       >   Unexpected arguments: FILENAME_VARIABLE;deploy_script
       > Call Stack (most recent call first):
       >   /nix/store/r7wdw4nkkjii690wwnzkgj68cg4shh2v-qtbase-6.10.0/lib/cmake/Qt6Core/Qt6CoreMacros.cmake:3910 (qt6_generate_deploy_app_script)
       >   bridge-gui/CMakeLists.txt:175 (qt_generate_deploy_app_script)
       >
       > 
       > -- Configuring incomplete, errors occurred!
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.17.2-zen1, NixOS, 25.11 (Xantusia), 25.11.20251025.6a08e6b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - nixpkgs: `/nix/store/jd3y5kfy3ywv9lc229nz8vxqbrcyv93d-source`

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/pr/protonmail-bridge-gui/package.nix
+++ b/pkgs/by-name/pr/protonmail-bridge-gui/package.nix
@@ -12,8 +12,12 @@
   gtest,
   sentry-native,
   protonmail-bridge,
+  fetchpatch2,
 }:
 
+let
+  guiSourcePath = "internal/frontend/bridge-gui";
+in
 stdenv.mkDerivation (finalAttrs: {
   pname = "protonmail-bridge-gui";
 
@@ -22,6 +26,18 @@ stdenv.mkDerivation (finalAttrs: {
   patches = [
     # Use `gtest` from Nixpkgs to allow an offline build
     ./use-nix-googletest.patch
+    # fix: qt6_generate_deploy_app_script deprecated keyword
+    (fetchpatch2 {
+      url = "https://github.com/daniel-fahey/proton-bridge/commit/9b282369523ce2aabcf95369a2ed3867efb5a4ac.patch?full_index=1";
+      relative = guiSourcePath;
+      hash = "sha256-CGBelsplAqLI+6GtDR+VELfVsLoHPN0324gfZ3obKpI=";
+    })
+    # fix: make libQt6WaylandEglClientHwIntegration conditional for Qt 6.10
+    (fetchpatch2 {
+      url = "https://github.com/daniel-fahey/proton-bridge/commit/7823fe4904186253798fc633724988d43dd642e0.patch?full_index=1";
+      relative = guiSourcePath;
+      hash = "sha256-2heMfZdXLL8/uyM0jZ860Jw76c+gHtc0G6pWmIRCONY=";
+    })
   ];
 
   nativeBuildInputs = [
@@ -44,7 +60,7 @@ stdenv.mkDerivation (finalAttrs: {
     sentry-native
   ];
 
-  sourceRoot = "${finalAttrs.src.name}/internal/frontend/bridge-gui";
+  sourceRoot = "${finalAttrs.src.name}/${guiSourcePath}";
 
   postPatch = ''
     # Bypass `vcpkg` by deleting lines that `include` BridgeSetup.cmake

